### PR TITLE
tests: bump mysql-tester version and update result

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #45961

Problem Summary:

### What is changed and how it works?

After mysql-tester commit https://github.com/pingcap/mysql-tester/pull/105
Several bugs are found and the test result need to update.

### Check List

Tests 

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/r/explain-non-select-stmt.result [hunk 1]
```diff
@@ -12,17 +12,15 @@ Insert	N/A	root		N/A
 explain format = 'brief' delete from t where a > 100;
 id	estRows	task	access object	operator info
 Delete	N/A	root		N/A
-└─SelectLock	3333.33	root		for update 0
```
Comment (tiancaiamao) [inline tests/integrationtest/r/explain-non-select-stmt.result:15]: The is bug of the old mysql-tester, there should not be a SelectLock unless in the 'select ... for update' statement.

File: tests/integrationtest/r/explain_easy.result [hunk 2]
```diff
@@ -55,14 +55,13 @@ HashJoin	4166.67	root		left outer join, equal:[eq(explain_easy.t1.c2, explain_ea
 explain format = 'brief' update t1 set t1.c2 = 2 where t1.c1 = 1;
 id	estRows	task	access object	operator info
 Update	N/A	root		N/A
-└─Point_Get	1.00	root	table:t1	handle:1, lock
```
Comment (tiancaiamao) [inline tests/integrationtest/r/explain_easy.result:58]: ditto.

File: tests/integrationtest/r/expression/issues.result [hunk 3]
```diff
@@ -2075,7 +2075,7 @@ EXECUTE mystmt USING @a;
 id	a
 select @@last_plan_from_cache;
 @@last_plan_from_cache
-0
+1
```
Comment (tiancaiamao) [inline tests/integrationtest/r/expression/issues.result:2078]: This should be the correct behaviour --- plan cache been used.

File: tests/integrationtest/r/planner/cascades/integration.result [hunk 4]
```diff
@@ -896,7 +896,7 @@ Projection_3	10000.00	root		Column#2, Column#4, Column#5, Column#6, Column#7, Co
 └─MemTableScan_4	10000.00	root	table:PROCESSLIST	
 select USER, DB, COMMAND, TIME, STATE, INFO, `DIGEST` from information_schema.processlist;
 USER	DB	COMMAND	TIME	STATE	INFO	DIGEST
-root	planner__cascades__integration	Query	0	in transaction; autocommit	select USER, DB, COMMAND, TIME, STATE, INFO, `DIGEST` from information_schema.processlist	78f1c0b2fcd082b6504b8ba48706391f13aad730a9efb7ec295688010bfec477
```
Comment (tiancaiamao) [inline tests/integrationtest/r/planner/cascades/integration.result:899]: Aha, catch the bug!

File: tests/integrationtest/r/planner/core/integration.result [hunk 5]
```diff
@@ -228,20 +228,6 @@ A	1
 Y	1
 a	1
 y	1
-create table strlist(a varchar(10) charset utf8mb4 collate utf8mb4_general_ci, b int) partition by list columns (a) (
```
Comment (tiancaiamao) [inline tests/integrationtest/r/planner/core/integration.result:231]: See here https://github.com/pingcap/tidb/pull/47358/files#diff-77ff1acea637e7d56230082a1f744f3454394f05500d2b6cc9f52710e8671c37R175

File: tests/integrationtest/r/planner/core/issuetest/planner_issue.result [hunk 6]
```diff
@@ -111,11 +111,12 @@ create table tbl_39(col_239 year(4) not null default '2009', primary key(col_239
...
 explain select  /*+ use_index_merge( tbl_39) */ col_239  from tbl_39 where not( tbl_39.col_239 not in ( '1994' ) ) and tbl_39.col_239 not in ( '2004' , '2010' , '2010' ) or not( tbl_39.col_239 <= '1996' ) and not( tbl_39.col_239 between '2026' and '2011' ) order by tbl_39.col_239 limit 382;
 id	estRows	task	access object	operator info
-Limit_12	382.00	root		offset:0, count:382
-└─IndexReader_39	382.00	root		index:Limit_38
-  └─Limit_38	382.00	cop[tikv]		offset:0, count:382
-    └─Selection_33	382.00	cop[tikv]		or(and(eq(planner__core__issuetest__planner_issue.tbl_39.col_239, 1994), not(in(planner__core__issuetest__planner_issue.tbl_39.col_239, 2004, 2010, 2010))), and(gt(planner__core__issuetest__planner_issue.tbl_39.col_239, 1996), or(lt(cast(planner__core__issuetest__planner_issue.tbl_39.col_239, double UNSIGNED BINARY), 2026), gt(cast(planner__core__issuetest__planner_issue.tbl_39.col_239, double UNSIGNED BINARY), 2011))))
-      └─IndexRangeScan_32	477.50	cop[tikv]	table:tbl_39, index:PRIMARY(col_239)	range:[1994,1994], (1996,+inf], keep order:true, stats:pseudo
+Projection_8	382.00	root		planner__core__issuetest__planner_issue.tbl_39.col_239
```
Comment (tiancaiamao) [inline tests/integrationtest/r/planner/core/issuetest/planner_issue.result:114]: The result before and after are both correct, it seems the 'tidb_partition_prune_mode' differ.

File: tests/integrationtest/r/privilege/privileges.result [hunk 7]
```diff
@@ -192,29 +192,30 @@ GRANT INSERT ON privilege__privileges.* TO  'tester'@'localhost';
 GRANT DROP ON privilege__privileges.* TO  'tester'@'localhost';
 alter table pt exchange partition p0 with table nt;
 CREATE USER 'test_import_into'@'localhost';
+GRANT SELECT ON privilege__privileges.* TO 'test_import_into'@'localhost';
 CREATE TABLE IF NOT EXISTS t(a int);
 IMPORT INTO t FROM '/file.csv';
-Error 1105 (HY000): cannot run IMPORT INTO in explicit transaction
```
Comment (tiancaiamao) [inline tests/integrationtest/r/privilege/privileges.result:197]: The old error is wrong ... catch the bug!

File: tests/integrationtest/t/planner/core/integration.test [hunk 8]
```diff
@@ -172,15 +172,16 @@ create table strrange(a varchar(10) charset utf8mb4 collate utf8mb4_general_ci,
...
-create table strlist(a varchar(10) charset utf8mb4 collate utf8mb4_general_ci, b int) partition by list columns (a) (
-						partition p0 values in ('a', 'b'),
-						partition p1 values in ('c', 'd'),
-						partition p2 values in ('e', 'f'));
-insert into strlist values ('a', 1), ('A', 1), ('d', 1), ('D', 1), ('e', 1);
---sorted_result
-select * from strlist where a='a';
---sorted_result
-select * from strlist where a in ('D', 'e');
+# TODO: need to run with collation enabled.
```
Comment (YangKeao) [inline tests/integrationtest/t/planner/core/integration.test:175]: Don't need to comment out the test. Run `./run-tests.sh` with `-d y` can generate the result file for it and pass the CI.

**Review by YangKeao — COMMENTED** [review]
Rest LGTM

File: tests/integrationtest/t/planner/core/integration.test (hunk 8 above)
Comment (Defined2014) [inline tests/integrationtest/t/planner/core/integration.test:175]: The orignal way is right. CI only check test cases with `new_collation_enable = false` when file name begins with `collation_`, like `collation_agg_func.test`

File: tests/integrationtest/t/planner/core/integration.test [hunk 9]
```diff
@@ -2040,6 +2041,7 @@ CREATE TABLE t(col1 enum('p5', '9a33x') NOT NULL DEFAULT 'p5',col2 tinyblob DEFA
 drop table if exists t;
 create table t (a int, b int, c int, index ia(a));
 select  * from t t1 join t t2 on t1.b = t2.b join t t3 on t2.b=t3.b join t t4 on t3.b=t4.b where t3.a=1 and t2.a=2;
+# TODO: The result is correct but does not exactly same.
 select plan from information_schema.statements_summary where SCHEMA_NAME = 'test' and STMT_TYPE = 'Select';
```
Comment (Defined2014) [inline tests/integrationtest/t/planner/core/integration.test:2045]: I think the schema_name should be `planner__core__integration` instead of test.

Comment (tiancaiamao) [reply tests/integrationtest/t/planner/core/integration.test]: I check the code and find this test can't be done in integration test.
The origical code is here https://github.com/pingcap/tidb/pull/41461/files#diff-fa42dbc5f88d5aef4ee1cb7452ee4497fc545324b76bee0e44f48e5f1af9f512R5213-R5251

File: tests/integrationtest/t/planner/core/integration.test (hunk 8 above)
Comment (tiancaiamao) [reply tests/integrationtest/t/planner/core/integration.test]: Done

Comment (tiancaiamao) [reply tests/integrationtest/t/planner/core/integration.test]: Done

**Comment by Defined2014** [comment]
@tiancaiamao should fix the `check_dev` and `check_dev_2`

We could use `./run-tests.sh -r all` to generate all results.

File: tests/integrationtest/r/planner/core/integration_partition.result [hunk 10]
```diff
@@ -739,29 +739,39 @@ a	b	c
 10	2022-01-01 00:00:00	Wowe
 explain format = 'brief' select * from t where a = 10 and b = "2022-01-01" and c = "Wow";
 id	estRows	task	access object	operator info
-  └─TableFullScan	12.00	cop[tikv]	table:t	keep order:false
-TableReader	0.52	root	partition:p3	data:Selection
-└─Selection	0.52	cop[tikv]		eq(rangecolumnsmulti.t.a, 10), eq(rangecolumnsmulti.t.b, 2022-01-01 00:00:00.000000), eq(rangecolumnsmulti.t.c, "Wow")
+      └─TableFullScan	10000.00	cop[tikv]	table:t	keep order:false, stats:pseudo
```
Comment (Defined2014) [inline tests/integrationtest/r/planner/core/integration_partition.result:742]: The result looks strange here. We shoud remove `--sorted_result` in `integration_partition.test`

Comment (tiancaiamao) [reply tests/integrationtest/r/planner/core/integration_partition.result]: Done

**Review by Defined2014 — APPROVED** [review]

File: tests/integrationtest/r/planner/core/integration_partition.result [hunk 11]
```diff
@@ -819,33 +835,33 @@ select * from t where a = 2 and b = "2022-01-02" and c < "Hi";
 a	b	c
 select * from t where a < 2;
 a	b	c
--2147483648	0000-00-00 00:00:00	
 -2147483648	0000-00-00 00:00:00	NULL
```
Comment (YangKeao) [inline tests/integrationtest/r/planner/core/integration_partition.result:838]: It seems that this result is unstable no matter with / without `--sorted_result`. Maybe it's a bug of `mysql-tester`? Let me take a look 🤔 .

**Review by YangKeao — APPROVED** [review]
Rest LGTM

File: tests/integrationtest/r/planner/core/integration_partition.result (hunk 11 above)
Comment (tiancaiamao) [reply tests/integrationtest/r/planner/core/integration_partition.result:838]: https://github.com/pingcap/mysql-tester/pull/108/

File: tests/integrationtest/run-tests.sh [hunk 12]
```diff
@@ -88,7 +88,7 @@ function build_mysql_tester()
 {
     echo "building mysql-tester binary: $mysql_tester"
     rm -rf $mysql_tester
-    GOBIN=$PWD go install github.com/pingcap/mysql-tester/src@da61c204fac56696f171298126b40c9277119223
+    GOBIN=$PWD go install github.com/pingcap/mysql-tester/src@5174bd1533c5a143ec1775ac1971f50b82a08539
```
Comment (Defined2014) [inline tests/integrationtest/run-tests.sh:91]:
```suggestion
    GOBIN=$PWD go install github.com/pingcap/mysql-tester/src@f3b554899a0cb24a3b387f7d45de2687098577c3
```

**Comment by YangKeao** [comment]
`executor/write` failed, but I cannot reproduce on my local machine :thinking:

```
run test [executor/write] err: sql:update t set id=id+17 where id in (3,10);: failed to run query 
"update t set id=id+17 where id in (3,10);" 
 around line 612, 
we need(99):
update t set id=id+17 where id in (3,10);
Error 1526 (HY000): Table has no partition for value 27
s
but got(99):
update t set id=id+17 where id in (3,10);
Error 1062 (23000): Duplicate entry '20' for key 't.idx'
```

/retest

**Comment by Defined2014** [comment]
> `executor/write` failed, but I cannot reproduce on my local machine 🤔
> 
> ```
> run test [executor/write] err: sql:update t set id=id+17 where id in (3,10);: failed to run query 
> "update t set id=id+17 where id in (3,10);" 
>  around line 612, 
> we need(99):
> update t set id=id+17 where id in (3,10);
> Error 1526 (HY000): Table has no partition for value 27
> s
> but got(99):
> update t set id=id+17 where id in (3,10);
> Error 1062 (23000): Duplicate entry '20' for key 't.idx'
> ```
> 
> /retest

A new PR merged. https://github.com/pingcap/tidb/pull/47420

**Pull request merged**